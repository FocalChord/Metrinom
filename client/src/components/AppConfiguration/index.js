import React from "react";
import { CssBaseline } from "@material-ui/core";
import { LastLocationProvider } from "react-router-last-location";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider, createMuiTheme, responsiveFontSizes } from "@material-ui/core/styles";
import { MetrinomProvider } from "../../context/MetrinomContext";

const darkTheme = responsiveFontSizes(
    createMuiTheme({
        palette: {
            type: "dark",
        },
    }),
);

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 *
 * These configurations are
 *  - ThemeProvider (Material UI Context which gives the color of the overall look and feel)
 *  - React Router
 *  - MetrinomProvider (Handles Loading of the Page)
 *  - Last Location Provider (Handles getting the last location url we were on)
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={darkTheme}>
            <CssBaseline />
            <MetrinomProvider>
                <Router>
                    <LastLocationProvider>{children}</LastLocationProvider>
                </Router>
            </MetrinomProvider>
        </ThemeProvider>
    );
};

export default AppConfiguration;
