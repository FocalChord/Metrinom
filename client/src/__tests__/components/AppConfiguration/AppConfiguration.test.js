import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AppConfiguration from "../../../components/AppConfiguration";

test("<AppConfiguration />", () => {
    const wrapper = shallow(
        <AppConfiguration>
            <div />
        </AppConfiguration>,
    );
    const tree = renderer
        .create(
            <AppConfiguration>
                <div />
            </AppConfiguration>,
        )
        .toJSON();

    expect(wrapper).toContainExactlyOneMatchingElement("ThemeProvider");
    expect(wrapper).toContainExactlyOneMatchingElement("MetrinomProvider");
    expect(wrapper).toContainExactlyOneMatchingElement("BrowserRouter");
    expect(tree).toMatchSnapshot();
});
