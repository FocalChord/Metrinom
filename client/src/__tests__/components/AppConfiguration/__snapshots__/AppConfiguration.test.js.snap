// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppConfiguration /> 1`] = `<div />`;
